### ISSUE TITLE
[$500] Random empty chats are showing in LHN in focus mode

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#38368]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @davidcardoza 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1709737869001459

## Action Performed:
1. Open app and sign in with an account with preference set `focus`
2. Look at the LHN

## Expected Result:
Should not display any random emptychats

## Actual Result:
10s of random empty state chats in the LHN, with no way to remove the chats from the LHN view
Note: They seems to be the members of the workspace room I am in

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![image (1)](https://github.com/Expensify/App/assets/38435837/df7a971c-1f31-47ec-abd6-8e91608a2bb4)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011e45e5cf4631d0fd</li>
        <li>Upwork Job ID: 1768690919242297344</li>
        <li>Last Price Increase: 2024-03-22</li>
    </ul>
</details>

**Comment by CortneyOfstad** [comment]
I was also able to recreate this yesterday — getting eyes on this!

**Comment by s77rt** [comment]
@tienifr Thanks for the proposal. Your RCA is not correct.

> the "unread" report will always show in LHN, even if it's empty

An empty report should not be considered as unread in the first place (there is nothing unread about it)

**Comment by jhohlfeld** [comment]
@m-natarajan I have a bit of a hard time reproducing this issue. Do you mind sharing a little more about how you achieved this?

Here's what I did to reproduce this issue:

1. login
2. go to profile -> preferences -> priority mode
3. switch priority mode to "#focus"
4. return to chat

I do have 3-4 chats with other sub-accounts which are empty. As expected, they are not showing up.
Only if I create a new chat I see this one empty. As expected, it is hidden as soon as I leave it.

App Version 1.4.53-2

**Comment by jhohlfeld** [comment]
Contributor details
Your Expensify account email: expensify@netronaut.de
Upwork Profile Link: https://www.upwork.com/freelancers/~0137c55e8ae1d137a8

**Comment by s77rt** [comment]
@tienifr Thanks for the update. The RCA makes sense and so the solution. Is there a specific case where `lastReadTime` is empty? (I'm not able to reproduce the bug)

**Comment by s77rt** [comment]
Not really able to reproduce, always getting the `lastReadTime` same as the `lastVisibleActionCreated`.

Regarding the unread logic, I have asked in Slack to confirm the expected behaviour in such cases https://expensify.slack.com/archives/C01GTK53T8Q/p1710865076998169

**Comment by s77rt** [comment]
Yes, I even tried creating a chat while signed out. After signing in I still find `lastReadTime` set. Probably getting the report via pusher, not sure why the last read time is set though. In any case we shall confirm the unread expected behaviour maybe it's expected that new chats appear as unread? ([slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1710865076998169))

**Comment by s77rt** [comment]
@tienifr Often the best thing to do is nothing. Perhaps this is no longer reproducible.

**Comment by s77rt** [comment]
@m-natarajan Can you please retest if this is still reproducible from your end?

**Comment by CortneyOfstad** [comment]
@m-natarajan, based on the comment [here](https://github.com/Expensify/App/issues/38368#issuecomment-2013736529), can you test to see if this is still reproducible by EOD? Thanks!

**Comment by CortneyOfstad** [comment]
@m-natarajan bump on the above ^^ thanks!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by s77rt** [comment]
Can we close this for now and reopen if this resurface again?

**Comment by CortneyOfstad** [comment]
Sounds good @s77rt 👍

**Comment by s77rt** [comment]
Agreed. Let's reopen this

**Comment by CortneyOfstad** [comment]
Sorry about that! Reopening now!

**Comment by s77rt** [comment]
@tienifr Let's get this fixed

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/38368#issuecomment-2001931057)

**Comment by flodnv** [comment]
>There're other ways to check if the report is empty as well, like checking the length of the chat's reportActions, or check if the reportActions contains only the report creation action.

That sounds better to me (we can say the chat is empty if it has 0 or 1 reportActions). Maybe we should do this same update in that other spot?

>isInGSDMode

I'm not sure why we named this like that, but seeing as it's called "focus" mode through the stack, let's name this to `isInFocusMode`

---

The proposal sounds fine, but why is the API returning these empty chats? Or is it that these empty chats were present locally in Onyx before switching to focus mode?

**Comment by s77rt** [comment]
Not overdue. Issue was just reopened. Almost there

**Comment by twisterdotcom** [comment]
Also experiencing this, would love it fixed. Are we sure this isn't a regression somehow? cc @puneetlath I feel like in the past you had worked on this.

**Comment by CortneyOfstad** [comment]
To keep this moving @s77rt can you provide an update on the PR by EOD today? Thanks!

**Comment by s77rt** [comment]
PR is ready for merge

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by CortneyOfstad** [comment]
PR is still in the review process.

**Comment by quinthar** [comment]
If we can't reproduce the issue, should we close this?

**Comment by s77rt** [comment]
This one is not easily reproducible, we closed it before but a got a [new one](https://github.com/Expensify/App/issues/39558) with same root cause. Thus this has been reopened (to handle the other case).

**Comment by CortneyOfstad** [comment]
Plan is to be merged on Monday after the merge freeze 🎉

**Comment by CortneyOfstad** [comment]
@s77rt can you have the checklist completed ASAP, so this does not delay payment? Thanks!

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: https://github.com/Expensify/App/pull/13419
- The offending PR has been commented on: https://github.com/Expensify/App/pull/13419#discussion_r1602425171
- A discussion in #expensify-bugs has been started: Not needed, unique bug
- Determine if we should create a regression test for this bug: No, the bug is hard to reproduce and shouldn't resurface

**Comment by CortneyOfstad** [comment]
## Payment Summary

@s77rt — paid $500 via Upwork
@tienifr — paid $500 via Upwork

Since a regression test is not needed, this can be closed. Thanks!
